docs/design/navigation-and-scale.md

```diff
@@ -0,0 +1,143 @@
+
+Menu follows standard responsiveness practises where it folds into icons and
+hamburger menu when space is limited.
+
+![Mobile navigation](mockups/23-03-2016-scale-and-navigation/mobile-navigation.png)
+
+## View templates
+
+This section shows concepts used as templates for all views.
+
+View which does not deal with a resource or a resource list, e.g., landing
+page which shows an overview of a cluster. The page consists of, potentially
+dismissable, blocks that display data.
+![Landing page](mockups/23-03-2016-scale-and-navigation/landing-page.png)
+
+Category view which shows lists of resources. In this mockup it is
+(not defined yet) "Apps" category which was selected through the
+navigation menu.
+![Category page template](mockups/23-03-2016-scale-and-navigation/category-page-template.png)
+
+Action bar replaces floating action button. The reason for this change
+is the fact that floating action buttons do not work well on desktop
+form factors, which are used by vast majority of our users. On the other
+hand, action bars work well on any form factors (desktop, mobile)
+and can scale for different actions on different resources.
+![Action bar](mockups/23-03-2016-scale-and-navigation/action-bar.png)
+
+Details of the resource list block:
+![Resource list block](mockups/23-03-2016-scale-and-navigation/resource-list-block.png)
+
+While the concept of cards used for displaying resources has some nice
+characteristics (like information density) it has been a constant source of
+problems. Users and developers complained about many issues: how do you sort
+cards? how do you change their order? how do you paginate them? how do you
+compare resources against each other? how do you distinct different resources
+in a card stream? how do you do bulk operations?
+
+We have identified that in order to solve these problems the cards can evolve
+into tabularized list items which can show exactly the same information as
+cards. This solution has many advantages, some of which are: pagination,
+ordering, bulk actions or visual comparison. Following mockup outlines
+the evolution:
+![Card evolution](mockups/23-03-2016-scale-and-navigation/card-evolution.png)
+
+Category views link to single resource views which show an unbounded list
+of resources of single type (e.g., Replica Sets) and a short summary/
+call to action block.
+![Single resource page template](mockups/23-03-2016-scale-and-navigation/single-resource-page-template.png)
+
+Details of an individual resource instance are shown on a views which displays
+its properites and all related objects (e.g., Services that target a Pod).
+![Details page template](mockups/23-03-2016-scale-and-navigation/details-page-template.png)
+
+## Concrete pages
+This section shows how templates proposed in the "View templates" section
+can be applied to real examples.
+
+"Apps" category:
+![Apps category](mockups/23-03-2016-scale-and-navigation/apps-category.png)
```

Comment:
> I'd eliminate endpoints, but we should think about where we could provide links to the app, that would hit external IPs/ports of the service, or would use the apiserver proxy.
 

Yeah, endpoints are problematic, but, in reality, very useful. Perhaps this needs better title/wording/documentation. Perhaps this can be moved to the unstructured part of the card.

But this is a problem with the specific page, not the proposed new UX language here. I assume non blocker.